components/merkledb/src/views/metadata.rs

```diff
@@ -344,6 +345,19 @@ impl<T: RawAccess> IndexesPool<T> {
         self.set_len(len + 1);
         (metadata, is_phantom)
     }
+
+    pub(crate) fn suffixes<K: BinaryKey + ?Sized>(&self, prefix: &str) -> Vec<K::Owned> {
+        let prefix = concat_keys!(prefix, INDEX_NAME_SEPARATOR);
+        let mut keys = Vec::new();
+
+        for (key, _) in self.0.iter::<_, Vec<u8>, Vec<u8>>(&prefix) {
```

Comment:
Using `bytes_iter` instead of `iter` here would save `Vec` allocations. (Don't know whether the compiler is smart enough to remove them, but I wouldn't rely on this fact.)